packages/flutter/lib/src/foundation/observer_list.dart

```diff
@@ -43,12 +43,15 @@ class ObserverList<T> extends Iterable<T> {
   ///
   /// Returns whether the item was present in the list.
   bool remove(T item) {
-    _isDirty = true;
-    _set.clear(); // Clear the set so that we don't leak items.
-    return _list.remove(item);
+    final bool removed = _list.remove(item);
+    if (removed) {
+      _isDirty = true;
+      _set.clear(); // Clear the set so that we don't leak items.
+    }
+    return removed;
   }
 
-  /// Removes all items from the list.
+  /// Removes all items from the [HashedObserverList].
```

Comment:
This is `ObserverList.clear`, I think you mean to change the documentation of `HashedObserverList.clear`?

Reply:
Indeed. Done